### ISSUE TITLE
Thermo: mixing scheme requires unique entry_id

### ISSUE THREAD

**Issue by rkingsbury** [materialsproject/emmet#370]
The MP DFT mixing scheme requires all `ComputedEntry` to have unique `entry_id` in order to process them. Currently the `entry_id` is set equal to the `material_id`, which means that, for example, a GGA and a R2SCAN calculation for a particular material will have the same `entry_id`. It might be helpful to store some type of modified mpid such as `mp-1234-GGA` in entry.data to facilitate use of the mixing scheme in `ThermoBuilder`. Alternatively we could use these suffixed mpids as the `entry_id` in `entries.< functional>` by default, without changing the `material_id` in the task document.

Flagging @arosen93 and @munrojm in case they have thoughts.

**Comment by munrojm** [comment]
I am not opposed to having `entry_id` formatted like that. One question, though. Currently, `run_type` in `entry.data` indicates the calculation type. Would using that plus the MPID not work?

**Comment by rkingsbury** [comment]
> One question, though. Currently, `run_type` in `entry.data` indicates the calculation type. Would using that plus the MPID not work?

@munrojm yes, that would work, with a caveat. For the mixing code _in pymatgen_, I'd prefer to rely only on the `entry_id`, because any `ComputedEntry` is guaranteed to have one, whereas `entry.data.run_type` may only be present in calculations retrieved from MP. So to make the mixing code as robust and useful as possible for users preparing their own calculations, I only want to rely on the attribute that must be present.

That said, we could definitely just combine `entry_id` and `run_type` into a pseudo-entry_id formatted as above. I think the question is where we do that. I see two options:

1. In `ThermoBuilder`, we temporarily overwrite the `entry_id` with the modified one as described above (e.g., `mp-1234-GGA`), run the mixing scheme, then restore the original `entry_id` to the processed entry.
2. We permanently set the `entry_id` of the respective entries in the Materials Doc to, e.g. `mp-1234-GGA`.

Either approach will work fine for the purposes of implementing the mixing scheme. I see a slight advantage to Option 2 from a transparency standpoint: as more and more R2SCAN data becomes available via the API, it might be nice for the `entry_id` to signal the users about which functional was used for a particular calculation.

**Comment by munrojm** [comment]
Okay, makes sense. I also agree with you about #2. Before making the change, I would want to take an inventory of what other code (including in pymatgen) that interacts with entries might assume an MPID for `entry_id`.

**Comment by mkhorton** [comment]
I would probably err _against_ modifying the `entry_id` to containing the run_type.

Note there's already some user confusion on the identifiers in thermo, e.g. if data.task_id should refer to a `task_id` or the `material_id`. There's some discussion of this here: https://matsci.org/t/query-result-is-different-when-using-legacy-api-vs-new-api/40569

Another solution might be to use e.g. `entry_id` as `f{material_id}({task_id})`, and then store both `task_id` and `material_id` separately in the `data` dict?

**Comment by munrojm** [comment]
> Another solution might be to use e.g. `entry_id` as `f{material_id}({task_id})`, and then store both `task_id` and `material_id` separately in the `data` dict?

 I like this suggestion.

**Comment by tsmathis** [comment]
closing as stale, current codebase (plus db versions since this issue was opened) have `entry_id`s as `{material_id}-{run_type}`, with both `material_id` and `task_id` stored under `entry.data` for both the `materials` and `thermo` endpoints

can revisit if needed
